brainmatch/brainmatch.py

```diff
@@ -136,6 +136,9 @@ def compute_total_score(proj_features, contrib_data):
     contrib_git_skills = [int(s) for s in contrib_data[
         experience_git_skills_field].split(" ") if s.isdigit()][0]
 
+    # Take highest git level if more than one are given
+    proj_features[git_skills_label] = proj_features[git_skills_label][-1]
```

Comment:
How are the `git_skills_labels` ordered? What are your assumptions on this?

Reply:
They go from 0 to 3.

Reply:
> They go from 0 to 3.

Because the shell script sorts them or because your form outputs them that way?

Reply:
It comes from the form.

Reply:
> It comes from the form.

Then it well may be the case that other forms do not output a sorted list, and the scripts should do that job.

Reply:
I do not expect any form to go from 3 to 0, or to have them mixed.

Reply:
> I do not expect any form to go from 3 to 0, or to have them mixed.

I'd say that we cannot safely assume that it would be the case for all files.

With this change you are acknowledging that there will be multiple `git_skills` values. And hence, you are implicitly acknowledging that the values may not be considered incremental nor mutually exclusive.

So in that case, and as done with any other category, it is safer to compute all matches for the category, as suggested [here](https://github.com/brainhackorg/brainmatch/pull/13#pullrequestreview-523685284).

Reply:
@eurunuela can you please comment on this so that we can agree on how the `git_skills`  field should work and make the appropriate changes if necessary? We can discuss it off-line if you prefer.

Reply:
I think the easiest is to force the registration forms to only accept one value for this question in particular.

I see too many possible situations on this one and we cannot cover them all. So I think it's better to not make this change and assume we only have one value for `git_skills`.

Reply:
In any case, I wanted to have a PR with the changes I had to make for the algorithm to work for the needs we had in Donostia.

Reply:
Again, this has to be kept general for other sites' needs, and as said, we cannot assume a given ordering. I am very willing to get this in and do the necessary changes to the Python script, but we need to agree on whether rather than assuming exclusive `git_skills` we will go for multiple values and summing them all to compute the score, as with the rest of the labels. Otherwise, the sorting can be done in the Python script with some additional work.

If people agree on a criteria that matches, and the standard registration template allows for multiple git skills values by default use it, then I'd be happy to commit that to this PR, adding the corresponding explanation to the `README` and adding the test cases.

Reply:
So for now the participants data will be assume to contain a single value for the `git_skills:` for each. Given that some projects do have already multiple `git_skills:` values, commit 8cbb140 sorts them in ascending order to keep the most demanding one.